### DISCUSSION TITLE
Feature request for SQL password Reset and API

### DISCUSSION THREAD

**Discussion by patstr81** [umbraco/Umbraco.Cloud.Issues#814] [Ideas]
### Issue description

Hi,

A brief history of my request. 
I reached out to Umbraco Support, as I wanted to be able to reset the password for the project SQL database. The reason for this is to be able to grant developers temporary access to directly connect to the database, and afterwards, just reset the password to prevent further access. (Yes using the whitelisting back and forth instead, would be an option)
I was told this was not possible to reset the password but that I could make a feature request instead.  
So to my requests.

**Request 1:**
Implement a feature to reset the password for the SQL connection for each projects database.

However, I would like to see this going one step further, to create possibilities to automate flows with Umbraco Cloud. This leads me to my second request. 

**Request 2:**
I would like to have an API created containing all the features that are available in the Umbraco Cloud UI, and the possibility to reset the SQL password this way as well. 
The API, should contain further feature like add/remove hostname, list projects and get the statistics in this list like bandwidth, subscription plan, etc. 
Also list the team members per project and the cloud. 
Show logs. 
The list of API features can be massive, and I can think of many features I would like to see in an API. 

The benefit of an API means it can be automated. Like Secret Server can be used to grant temporary access to the SQL database for developers. 
As you don't have a granular way of giving access to certain features in Umbraco cloud, having a an API would leave that up to the customers which API certain individuals can use e.g. from an website form or application. E.g. I can create an application where deliver leads have access to add hostnames, or to add users to backoffice. 

Would this be viable?

Regards
Patrik

**Comment by sajumb** [comment]
Hi @patstr81,
Thank you for your suggestions regarding enhancements to Umbraco Cloud. We appreciate your interest in adding a feature to reset the SQL database password for individual projects, and your proposal for a comprehensive API that would extend the functionalities available in the Umbraco Cloud UI.

We acknowledge the potential benefits of these features, particularly in terms of automation and enhanced project management. However, while we find these ideas intriguing, we have not yet commenced development on these specific functionalities.

Regarding the implementation of a public endpoint for SQL database password reset and the comprehensive API, these suggestions have been noted and are under consideration. They have been included in our internal discussions for future improvements. However, at this stage, these features are not part of our immediate development roadmap. We are carefully evaluating our resources and priorities to ensure that we continue to provide the best possible service to our users.

We understand the importance of these features to your workflow and will keep your suggestions in mind as we plan our future development.
